File: integration-test/java/spring-app/src/main/java/com/qualityminds/lazyval/integration/boundary/persistence/mongodb/SpringDataMongoRepository.java
package com.qualityminds.lazyval.integration.boundary.persistence.mongodb;

import com.qualityminds.lazyval.integration.shared.Isbn;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface SpringDataMongoRepository extends MongoRepository<MongoOrder, UUID> {

    @Query("{ 'isbn' : ?0 }")
    List<MongoOrder> findByIsbn(Isbn isbn);
}

File: integration-test/kotlin/spring-app/src/main/kotlin/com/qualityminds/lazyval/integration/boundary/persistence/mongodb/SpringDataMongoRepository.kt
package com.qualityminds.lazyval.integration.boundary.persistence.mongodb

import com.qualityminds.lazyval.integration.shared.Isbn
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface SpringDataMongoRepository : MongoRepository<MongoOrder, UUID> {
    @Query("{ 'isbn' : ?0 }")
    fun findByIsbn(isbn: Isbn): MutableList<MongoOrder>?
}